src/Semantic/Graph.hs

```diff
@@ -301,36 +299,32 @@ parsePythonPackage parser project = do
       resMap <- Task.resolutionMap project
       pure (Package.fromModules (Data.Abstract.Evaluatable.name (projectName project)) modules resMap) -- TODO: Confirm this is the right `name`.
     PythonPackage.Packages dirs -> do
-      filteredBlobs <- for dirs $ \dir -> do
-        let packageDir = projectRootDir project </> unpack dir
-        let paths = filter ((packageDir `isPrefixOf`) . filePath) (projectFiles project)
-        traverse (readFile project) paths
+      let filteredBlobs = do
+            dir <- dirs
+            let packageDir = projectRootDir project </> unpack dir
+            filter ((packageDir `isPrefixOf`) . blobPath) (projectBlobs project)
       packageFromProject project filteredBlobs
     PythonPackage.FindPackages excludeDirs -> do
       trace "In Graph.FindPackages"
       let initFiles = filter (("__init__.py" `isSuffixOf`) . filePath) (projectFiles project)
       let packageDirs = filter (`notElem` ((projectRootDir project </>) . unpack <$> excludeDirs)) (takeDirectory . filePath <$> initFiles)
-      filteredBlobs <- for packageDirs $ \dir -> do
-        let paths = filter ((dir `isPrefixOf`) . filePath) (projectFiles project)
-        traverse (readFile project) paths
+      let filteredBlobs = do
+            dir <- packageDirs
+            filter ((dir `isPrefixOf`) . blobPath) (projectBlobs project)
       packageFromProject project filteredBlobs
     where
       packageFromProject project filteredBlobs = do
-        let p = project { projectBlobs = catMaybes $ join filteredBlobs }
+        let p = project { projectBlobs = filteredBlobs }
         modules <- fmap (fmap snd) <$> parseModules parser p
         resMap <- Task.resolutionMap p
         pure (Package.fromModules (Data.Abstract.Evaluatable.name $ projectName p) modules resMap) -- TODO: Confirm this is the right `name`.
 
-parseModule :: (Member (Error SomeException) sig, Member Parse sig, Carrier sig m)
+parseModule :: (Member Parse sig, Carrier sig m)
             => Project
             -> Parser term
-            -> File
+            -> Blob
             -> m (Module (Blob, term))
-parseModule proj parser file = do
-  mBlob <- readFile proj file
-  case mBlob of
-    Just blob -> moduleForBlob (Just (projectRootDir proj)) blob . (,) blob <$> parse parser blob
-    Nothing   -> throwError (SomeException (FileNotFound (filePath file)))
```

Comment:
Is this a _different_ `FileNotFound`? It didn’t look like the previous one was being exported. Is this one still being used anywhere?

Reply:
Oh it _was_ being exported, I just read it incorrectly because the constructor & type names were so different.